src/Analysis/Engine/Impl/Values/ProtocolInfo.cs

```diff
@@ -129,17 +129,15 @@ class ProtocolInfo : AnalysisValue, IHasRichDescription {
         }
 
 
-        internal override void AddReference(Node node, AnalysisUnit analysisUnit) {
-            _references.GetReferences(analysisUnit.ProjectEntry as ProjectEntry)?.AddReference(new EncodedLocation(analysisUnit, node));
-        }
+        internal override void AddReference(Node node, AnalysisUnit analysisUnit)
+            => _references.GetReferences(analysisUnit.ProjectEntry as ProjectEntry)?.AddReference(new EncodedLocation(analysisUnit, node));
 
         public override IEnumerable<ILocationInfo> Locations {
             get {
-                ReferenceList defns;
-                if (!_references.TryGetValue(DeclaringModule, out defns)) {
+                if (!_references.TryGetValue(DeclaringModule, out var defns)) {
                     return Enumerable.Empty<ILocationInfo>();
                 }
-                return defns.Definitions.Select(l => l.GetLocationInfo()).Where(l => l != null);
+                return defns.Definitions.Select(l => l.GetLocationInfo()).ExcludeDefault();
```

Comment:
Looks like we have two almost identical methods: `ExcludeDefaults` and `WhereNotNull`.

Reply:
Yes `ExcludeDefault` came from RTVS extensions, looks like `WhereNotNull` is local thing